src/Highlight.js

```diff
@@ -0,0 +1,46 @@
+import React, { PropTypes, Component } from 'react';
+
+class Highlight extends Component {
+
+  getIndices() {
+    const search = this.props.search.toLowerCase();
+    const origName = this.props.name;
+    const name = origName.toLowerCase();
+    const idxArr = [];
+    const highlight = {};
+    for (let i = 0; i < search.length; i++) {
+      if (name[i] === search[i]) {
+        idxArr.push(i);
+      }
+    }
+    if (idxArr.length === 1) {
+      highlight.isHL = origName.slice(0,1);
+      highlight.noHL = origName.slice(1);
+    } else if (idxArr.length > 1) {
+      highlight.isHL = origName.slice(idxArr[0],idxArr[idxArr.length - 1] + 1);
+      highlight.noHL = origName.slice(idxArr[idxArr.length - 1] + 1);
+    } else {
+      highlight.noHL = origName;
+    }
+    return (highlight);
```

Comment:
When writing such complex logic I would always add one liners that describe what is being done in the code. A reader should always understand what the code does without reading the actual statements.